DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/board_test/src/usm_speed.hpp

```diff
@@ -0,0 +1,177 @@
+class USMMemCopy;
+class USMMemRead;
+class USMMemWrite;
+
+using long8 = sycl::vec<long, 8>;
+
+enum USMTest { MEMCOPY, READ, WRITE };
+constexpr size_t kIterations = 1;
+constexpr size_t kNumBytes = 1024 * 1024 * 1024;
+constexpr size_t kNumItems = kNumBytes / sizeof(long8);
+
+// MEMCOPY test: launches a kernel to copy data from one USM pointer to another.
+sycl::event memcopy_kernel(sycl::queue &q, long8 *in, long8 *out) {
+  return q.single_task<USMMemCopy>([=]() [[intel::kernel_args_restrict]] {
+    sycl::host_ptr<long8> in_h(in);
+    sycl::host_ptr<long8> out_h(out);
+    for (size_t i = 0; i < kNumItems; i++) {
+      out_h[i] = in_h[i];
+    }
+  });
+}
+
+// READ test: launches a kernel to read data from a USM pointer, sum it up, and
+// store to an output pointer.
+sycl::event read_kernel(sycl::queue &q, long8 *in, long8 *out) {
+  return q.single_task<USMMemRead>([=]() {
+    sycl::host_ptr<long8> in_h(in);
+    sycl::host_ptr<long8> out_h(out);
+    long8 sum{0};
+    for (size_t i = 0; i < kNumItems; i++) {
+      sum += in_h[i];
+    }
+    // This prevents the reads from being optimized away
+    out_h[0] = sum;
+  });
+}
+
+// WRITE test: launches a kernel to write data to a USM pointer.
+sycl::event write_kernel(sycl::queue &q, long8 *in, long8 *out) {
+  return q.single_task<USMMemWrite>([=]() {
+    sycl::host_ptr<long8> out_h(out);
+    long8 answer{5};
```

Comment:
the value 5 is hard coded and supplied in multiple places. I think it would be good if there was a constexpr const long that holds the value 5 and a comment stating that it's arbitrary